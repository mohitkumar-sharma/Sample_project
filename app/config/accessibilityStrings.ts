export default {
  NORMAL_BUTTON_TEXT: 'normal-button-text',
  SLIDER_BUTTON_TEXT: 'slider-button-text',
  NORMAL_BUTTON: 'normal-button',
  SLIDER_BUTTON_ICON: 'slider-button-icon',
  SLIDER_BUTTON_CONTAINER: 'slider-button-container',
  BANNER_TEXT: 'banner-text',
  HEADER_SIDES_TEXT: 'header-sides-text',
  HEADER_PRESSABLE_ELEMENT: 'header-pressable-element',
  SETTINGS_SWITCH_TEXT: 'settings-switch-text',
  SETTINGS_SWITCH: 'settings-switch',
  INV_TEXT_INPUT: 'inv-text-input',
};
