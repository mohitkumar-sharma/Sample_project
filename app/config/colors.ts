export default {
  black: '#000000',
  gray: '#9B9B9B',
  divider: '#E8E8E8',
  white: '#FFFFFF',
  themeBlue: '#6EB1F7',
  themeLightBlack: '#34424A',
  appTheme: '#1A1924',
  borderLine: '#59606A',
  warning: '#FFCC00',
  red: '#FF0000',
};
