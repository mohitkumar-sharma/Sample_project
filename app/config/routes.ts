export default {
  WELCOME: 'Welcome',
  HOME: 'Home',
  SETTINGS: 'Settings',
};
