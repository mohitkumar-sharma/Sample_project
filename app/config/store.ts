import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { rootReducer } from '../states';

const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware(),
});

export type AppDispatch = typeof store.dispatch;

export { store };
