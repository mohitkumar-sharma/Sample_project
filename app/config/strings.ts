export default {
  GO_TO_HOME: 'Go To Home',
  DIAMOND_ICON: 'diamond',
  SLIDE_ME: 'Slide me to continue',
  APP_RUNNING_WARNING: 'Notice: You are running this app on',
  WELCOME_INPUT_PLACEHOLDER: 'Enter your name here...',
  SIGN_OUT: 'SignOut',
  HOME_BUTTONS_HEADING: '4 variations of a button',
  PRESS_ME: 'Press me',
  TRANSPARENT_BTN_PRESSED: 'Transparent button is pressed',
  LIGHT_BLACK_BTN_PRESSED: 'Light black button is pressed',
  BLUE_BTN_PRESSED: 'Blue button is pressed',
  SWITCH_TEXT: 'Enable Warning Banner',
  ALERT_TITLE_INVALID_USER_NAME: 'Invalid UserName!',
  ALERT_TITLE_BTN_PRESSED: 'Button Pressed!',
  ALERT_ENTER_VALID_USER_NAME: 'Please enter a valid user name.',
};
