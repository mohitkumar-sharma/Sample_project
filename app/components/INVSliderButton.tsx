import React from 'react';
import { StyleSheet, StyleProp, TextStyle, Text, View, ViewStyle } from 'react-native';
import Slider from 'react-native-slide-to-unlock';
import ButtonSliderElement, { ButtonSliderElementProps } from './ButtonSliderElement';
import config from '../config';

export interface INVSliderButtonProps extends ButtonSliderElementProps {
  /* as an object of style props for TouchableOpacity element */
  containerStyle?: StyleProp<ViewStyle>;
  /* as a function triggered when sliding reached at end of Slider element */
  onEndReached?: () => void;
  /* as an string to show the text of button on Text element */
  buttonText?: string;
  /* as an object of style props for Text element */
  buttonTextStyle?: StyleProp<TextStyle>;
}

/**
 * This is the INVSliderButton functional component works as a custom slider button
 * @param params : params is an object of INVSliderButtonProps and ButtonSliderElementProps
 */
const INVSliderButton = ({
  buttonText,
  onEndReached,
  buttonTextStyle,
  containerStyle,
  ...props
}: INVSliderButtonProps): JSX.Element => {
  return (
    <View style={containerStyle} testID={config.accessibilityStrings.SLIDER_BUTTON_CONTAINER}>
      <Slider
        onEndReached={() => onEndReached?.()}
        containerStyle={styles.wrapper}
        sliderElement={<ButtonSliderElement {...props} />}>
        {buttonText && buttonText !== '' ? (
          <Text
            style={[styles.buttonText, buttonTextStyle]}
            testID={config.accessibilityStrings.SLIDER_BUTTON_TEXT}>
            {buttonText}
          </Text>
        ) : null}
      </Slider>
    </View>
  );
};

const styles = StyleSheet.create({
  wrapper: {
    height: 55,
    marginHorizontal: 20,
    borderRadius: 9,
    borderWidth: 1,
    borderColor: config.colors.themeBlue,
    justifyContent: 'center',
  },
  buttonText: {
    color: config.colors.themeBlue,
    fontSize: 17,
    fontWeight: config.fontWeights.FONT_WEIGHT_REGULAR,
  },
});

export default INVSliderButton;
