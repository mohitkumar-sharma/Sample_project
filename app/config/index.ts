import colors from './colors';
import localConfig from './localConfig';
import strings from './strings';
import routes from './routes';
import fontWeights from './fontWeights';
import accessibilityStrings from './accessibilityStrings';
import { store } from './store';

const config = {
  colors,
  ...localConfig,
  store,
  strings,
  routes,
  fontWeights,
  accessibilityStrings,
};

export default config;
