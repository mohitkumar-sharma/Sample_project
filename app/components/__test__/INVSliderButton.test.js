import React from 'react';
import { render } from '@testing-library/react-native';
import INVSliderButton from '../INVSliderButton';
import config from '../../config';

describe('INVSliderButton', () => {
  const props = {
    iconContainerStyle: { marginHorizonta: 20 },
    iconStyle: { fontSize: 20 },
    buttonTextStyle: { fontSize: 19 },
    containerStyle: { marginTop: 15 },
  };

  const iconName = config.strings.DIAMOND_ICON;
  const buttonText = config.strings.SLIDE_ME;
  const onEndReached = jest.fn();

  it('matches snapshot', () => {
    expect(render(<INVSliderButton {...props} />)).toMatchSnapshot();
    expect(
      render(
        <INVSliderButton
          {...props}
          iconName={iconName}
          buttonText={buttonText}
          onEndReached={onEndReached}
        />
      )
    ).toMatchSnapshot();
  });

  it('renders the button text', () => {
    const { queryByTestId } = render(<INVSliderButton {...props} buttonText={buttonText} />);
    expect(queryByTestId(config.accessibilityStrings.SLIDER_BUTTON_TEXT)).toBeTruthy();
  });

  it('does not render the button text', () => {
    const { queryByTestId } = render(<INVSliderButton {...props} />);
    expect(queryByTestId(config.accessibilityStrings.SLIDER_BUTTON_TEXT)).toBeFalsy();
  });

  it('does not call on sliding ends', () => {
    const { getByTestId } = render(<INVSliderButton {...props} />);
    const sliderButtonContainer = getByTestId(config.accessibilityStrings.SLIDER_BUTTON_CONTAINER);
    const [sliderButton] = sliderButtonContainer.children;
    expect(sliderButton.props.onEndReached()).toBeUndefined();
  });

  it('call on sliding ends', () => {
    const { getByTestId } = render(<INVSliderButton {...props} onEndReached={onEndReached} />);
    const sliderButtonContainer = getByTestId(config.accessibilityStrings.SLIDER_BUTTON_CONTAINER);
    const [sliderButton] = sliderButtonContainer.children;
    sliderButton.props.onEndReached();
    expect(onEndReached).toHaveBeenCalledWith();
  });
});
