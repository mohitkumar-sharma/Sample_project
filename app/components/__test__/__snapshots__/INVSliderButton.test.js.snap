// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`INVSliderButton matches snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 15,
    }
  }
  testID="slider-button-container"
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#6EB1F7",
          "borderRadius": 9,
          "borderWidth": 1,
          "height": 55,
          "justifyContent": "center",
          "marginHorizontal": 20,
        },
        Object {
          "alignItems": "flex-start",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#6EB1F7",
              "borderRadius": 8,
              "height": 55,
              "justifyContent": "center",
              "width": 55,
            },
            Object {
              "marginHorizonta": 20,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
            "zIndex": -1,
          },
          Object {},
        ]
      }
    />
  </View>
</View>
`;

exports[`INVSliderButton matches snapshot 2`] = `
<View
  style={
    Object {
      "marginTop": 15,
    }
  }
  testID="slider-button-container"
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "borderColor": "#6EB1F7",
          "borderRadius": 9,
          "borderWidth": 1,
          "height": 55,
          "justifyContent": "center",
          "marginHorizontal": 20,
        },
        Object {
          "alignItems": "flex-start",
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#6EB1F7",
              "borderRadius": 8,
              "height": 55,
              "justifyContent": "center",
              "width": 55,
            },
            Object {
              "marginHorizonta": 20,
            },
          ]
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FFFFFF",
                "fontSize": 20,
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 25,
                },
                Object {
                  "fontSize": 20,
                },
              ],
              Object {
                "fontFamily": "simple-line-icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="slider-button-icon"
        >
          ÓÅÉ
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignSelf": "center",
            "position": "absolute",
            "zIndex": -1,
          },
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#6EB1F7",
              "fontSize": 17,
              "fontWeight": "400",
            },
            Object {
              "fontSize": 19,
            },
          ]
        }
        testID="slider-button-text"
      >
        Slide me to continue
      </Text>
    </View>
  </View>
</View>
`;
